processor/src/fast/mod.rs

```diff
@@ -320,17 +323,113 @@ impl FastProcessor {
         self.execute_impl(program, host).await
     }
 
-    /// Executes the given program and returns the stack outputs.
-    ///
-    /// This function is mainly split out of `execute()` for testing purposes so that we can access
-    /// the internal state of the `FastProcessor` after execution.
     async fn execute_impl(
         &mut self,
         program: &Program,
         host: &mut impl AsyncHost,
     ) -> Result<StackOutputs, ExecutionError> {
-        self.execute_mast_node(program.entrypoint(), program.mast_forest(), program.kernel(), host)
-            .await?;
+        let mut continuation_stack = ContinuationStack::new(program);
+
+        while let Some((current_forest, processing_step)) = continuation_stack.pop_continuation() {
```

Comment:
Would it make sense to extract this into an `execute_step` function? This was something I think was discussed async yesterday.